yaksh/views.py

```diff
@@ -2100,13 +2098,18 @@ def update_email(request):
 
 @login_required
 @email_verified
-def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
+def download_assignment_file(request, quiz_id, course_id=None,
+                             question_id=None, user_id=None):
     user = request.user
-    if not is_moderator(user):
-        raise Http404("You are not allowed to view this page")
+    course = Course.objects.get(id=course_id)
```

Comment:
This will fail at some point, given that course_id can be none and get will fail!